src/main/scala/de/upb/cs/swt/delphi/instanceregistry/RequestHandler.scala

```diff
@@ -315,45 +315,53 @@ class RequestHandler(configuration: Configuration, authDao: AuthDAO, instanceDao
-
-            val traefikConfig = TraefikConfiguration(traefikHost, configuration.traefikUri)
-
-            val newInstance = Instance(Some(id),
-              host,
-              port,
-              name.getOrElse(s"Generic $componentType"),
-              componentType,
-              Some(dockerId),
-              InstanceState.Deploying,
-              List.empty[String],
-              List.empty[InstanceLink],
-              List.empty[InstanceLink],
-              Some(traefikConfig)
-            )
-
-            instanceDao.updateInstance(newInstance) match {
-              case Success(_) =>
-                log.info("Instance successfully registered.")
-                fireInstanceAddedEvent(newInstance)
-                fireNumbersChangedEvent(newInstance.componentType)
-                Success(id)
-              case Failure(x) =>
-                log.warning(s"Failed to register. Exception: $x")
-                Failure(x)
-            }
+        try {
+
+          val deployResult = Await.result(future, timeout.duration).asInstanceOf[Try[(String, String, Int, String)]]
+          deployResult match {
+            case Failure(ex) =>
+              log.warning(s"Failed to deploy container, docker host not reachable. Message ${ex.getMessage}")
+              instanceDao.removeInstance(id)
+              fireDockerOperationErrorEvent(None, s"Deploy failed with message: ${ex.getMessage}")
+              Failure(new RuntimeException(s"Failed to deploy container, docker host not reachable (${ex.getMessage})."))
+            case Success((dockerId, host, port, traefikHost)) =>
+              log.info(s"Deployed new $componentType container with docker id: $dockerId, host: $host, port: $port and Traefik host: $traefikHost.")
+
+              val traefikConfig = TraefikConfiguration(traefikHost, configuration.traefikUri)
+
+              val newInstance = Instance(Some(id),
+                host,
+                port,
+                name.getOrElse(s"Generic $componentType"),
+                componentType,
+                Some(dockerId),
+                InstanceState.Deploying,
+                List.empty[String],
+                List.empty[InstanceLink],
+                List.empty[InstanceLink],
+                Some(traefikConfig)
+              )
+
+              instanceDao.updateInstance(newInstance) match {
+                case Success(_) =>
+                  log.info("Instance successfully registered.")
+                  fireInstanceAddedEvent(newInstance)
+                  fireNumbersChangedEvent(newInstance.componentType)
+                  Success(id)
+                case Failure(x) =>
+                  log.warning(s"Failed to register. Exception: $x")
+                  Failure(x)
+              }
+          }
+        }
+        catch {
+          case ex: Exception => log.warning(s"Exception occured with the messege: $ex")
```

Comment:
I see two problems here:
* When deployment fails the instance needs to be removed from the dao (as seen in line 324)
* There's a spelling error, mess**e**ge should be mess**a**ge, also you can remove the word 'the'